tests/cpp/test_gpu3.cpp

```diff
@@ -8202,6 +8202,53 @@ TEST_F(NVFuserTest, ReverseMerge) {
   ASSERT_TRUE(t0.equal(cg_outputs.at(0)));
 }
 
+TEST_F(NVFuserTest, preallocated_outputs) {
+  // Fusion definition inspired from NVFuserTest.AsyncCompilation_CUDA
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* tv0 = makeSymbolicTensor(2);
+  TensorView* tv1 = makeSymbolicTensor(1);
+  TensorView* tv2 = makeSymbolicTensor(2);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addInput(tv2);
+
+  TensorView* tv3 = add(tv0, IrBuilder::create<Val>(1.0)); // Group 0
+  TensorView* tv4 = max(tv3, {0}); // Group 0
+  TensorView* tv5 = add(tv4, tv1); // Group 0
+  TensorView* tv6 = add(tv5, tv2); // Group 1
```

Comment:
I'm sure that you've carefully constructed the example so it triggers segmentation, and thanks for testing that. But segment_set is probably an easier and more future-proof way to trigger segmentation if that's your main concern.

Reply:
right, I did as you suggest and simplified the test. Thanks!